File: 69. Sqrt(x).py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        l = 0
        r = x/2 + 1
        while l <= r:
            mid = (l + r)/2
            if mid * mid == x:
                return mid
            elif mid * mid < x:
                l = mid + 1
            else:
                r = mid - 1
        return r

File: 69. Sqrt(x).cpp
class Solution {
public:
    int mySqrt(int x) {
        long long int l = 0;
        long long int r = x/2 + 1;
        while (l <= r)
        {
            long long int mid = (int)((l+r) / 2);
            if (mid*mid == x)
                return mid;
            else if (mid*mid < x)
                l = mid + 1;
            else
                r = mid - 1;
        }
        return r;
    }
};
